eventdemo/source/objc/EventDemo/ChatClientManager.m

```diff
@@ -0,0 +1,195 @@
+    }
+    return self;
+}
+
+- (void)initializeKaa
+{
+    //Create a Kaa client with the Kaa default context.
+    self.kaaClient = [KaaClientFactory
+                      clientWithContext:[[DefaultKaaPlatformContext alloc] init]
+                      stateDelegate:self];
+    [self.kaaClient setProfileContainer:self];
+    
+    // Start the Kaa client and connect it to the Kaa server.
+    [self.kaaClient start];
+    
+    //Attaching new user
+    [self.kaaClient attachUserWithId:USER_EXTERNAL_ID
+                         accessToken:USER_ACCESS_TOKEN delegate:self];
+}
+
+#pragma mark - Kaa Delegates methods
+
+#pragma mark - KaaClientStateDelegate
+
+- (void)onStarted
+{
+    NSLog(@"Kaa client started");
+}
+
+#pragma mark - UserAttachDelegate
+
+- (void)onAttachResult:(UserAttachResponse *)response
+{    
+    if (response.result == SYNC_RESPONSE_RESULT_TYPE_SUCCESS)
+    {
+        NSLog(@"Endpoint successfully attached!");
+        //Obtain the event family factory.
+        EventFamilyFactory *eventFamilyFactory = [self.kaaClient getEventFamilyFactory];
+        
+        //Obtain the concrete event family.
+        self.chatEventFamily = [eventFamilyFactory getChat];
+        [self.chatEventFamily addDelegate:self];
+    }
+    else
+    {
+        NSLog(@"Endpoint attach failed: event demo stopped");
+    }
+}
+
+#pragma mark - ChatDelegate
+
+- (void)onChatEvent:(ChatEvent *)event fromSource:(NSString *)source
+{
+    NSLog(@"Got chat event");
+    
+    //Handling chat event
+    switch (event.EventType)
+    {
+        case CHAT_EVENT_TYPE_CREATE:
+            //Handling CREATE chat event
+            [self createRoom:event.ChatName onlyLocal:YES];
+            break;
+            
+        case CHAT_EVENT_TYPE_DELETE:
+            //Handling CREATE chat event
+            [self deleteRoom:event.ChatName onlyLocal:YES];
+            break;
+    }
+}
+
+- (void)onMessage:(Message *)event fromSource:(NSString *)source
+{
+    NSLog(@"Got chat message");
+    [self saveMessage:event.Message toRoom:event.ChatName];
+}
+
+#pragma mark - ProfileContainer
+
+- (KAAEmptyData *)getProfile
+{
+    return [[KAAEmptyData alloc] init];
+}
+
+#pragma mark - Sending Kaa Events logic
+
+- (void)createRoom:(NSString *)name onlyLocal:(BOOL)onlyLocal
+{
+    if(name.isEmpty == NO) // chat room nema should not be empty
+    {
+        if(onlyLocal == NO) //Check if we don't need to save it only locally
+        {
+            //create and send CREATE chat event to the Kaa
+            ChatEvent *addEvent = [[ChatEvent alloc] initWithChatName:name
+                                                            EventType:CHAT_EVENT_TYPE_CREATE];
+            [_chatEventFamily sendChatEventToAll:addEvent];
+        }
+        [_rooms addObject:name];
+        [self notify:RoomsListUpdated];
+    }
+}
+
+- (void)deleteRoom:(NSString *)room  onlyLocal:(BOOL)onlyLocal
+{
+    if([_rooms containsObject:room])
```

Comment:
Insert a space between control structures if, while, for, return, etc. and their opening parenthesis